flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/assigners/MySqlSnapshotSplitAssigner.java

```diff
@@ -300,6 +300,7 @@ private void splitTable(TableId nextTable) {
                 remainingSplits.addAll(schemaLessSnapshotSplits);
                 if (!chunkSplitter.hasNextChunk()) {
                     remainingTables.remove(nextTable);
+                    addAlreadyProcessedTablesIfNotExists(nextTable);
```

Comment:
We should change the code in `captureNewlyAddedTables` instead of changing the location of `addAlreadyProcessedTablesIfNotExists`.

Reply:
Noted, I'll change the implementation asap